torch/utils/benchmark/utils/compile.py

```diff
@@ -0,0 +1,192 @@
+    Train/Inference | Backend        | Mode | Compilation Time | Average Running Time | Speedup   
+    ----------------+----------------+------+------------------+----------------------+-----------
+    Training        | Eager          | -    | -                |     0.0005           | -         
+    Training        | aot_ts_nvfuser | -    |     0.5023       |     0.0006           |    -0.3070
+    Training        | cudagraphs     | -    |     0.0242       |     0.0259           |   -51.4658
+
+    The important warnings are
+    Your model is loaded on GPU
+    Your GPU supports tensor cores, we will enable it automatically by setting `torch.set_float32_matmul_precision('high')`
+
+    If a compilation fails for any reason including the dependency not being included then we will print Failed to compile {backend} with mode {mode}
+
+
+    """
+    if next(model.parameters()).is_cuda:
+        print("Your model is loaded on GPU")
+        if torch.backends.cuda.matmul.allow_tf32 is False and torch.cuda.get_device_capability() >= (8, 0):
+            print("Your GPU supports tensor cores, we will enable it automatically by setting `torch.set_float32_matmul_precision('high')`")
+            torch.set_float32_matmul_precision("high")
+
+
+    table = SimpleTable(field_names = ["Train/Inference", "Backend", "Mode", "Compilation Time", "Average Running Time", "Speedup"])
+
+
+    eager_time = None
+    torchdynamo.reset()
+    _, eager_time = benchmark_compile(model, sample_input, num_iters, None, None, is_training, optimizer)
+    table.add_row([("Training" if is_training else "Inference"), "Eager", "-", "-", eager_time, "-"])
+
+    for backend in torchdynamo.list_backends():
+        if backend == "ipex":  # ipex has an annoying import error it prints
+            continue
+
+        if backend == "inductor":
+            for mode in list(torch._inductor.list_mode_options().keys()) + [None]:
+                if mode == "default":
+                    continue
+                torchdynamo.reset()
+                compilation_time, running_time = benchmark_compile(model, sample_input, num_iters, backend, mode, is_training, optimizer)
+                if running_time is not None:
+                    speedup = (eager_time - running_time) / eager_time if eager_time else None
+                    table.add_row([("Training" if is_training else "Inference"), backend, mode or "-", compilation_time or "-", running_time, speedup or "-"])
+        else:
+            torchdynamo.reset()
+            compilation_time, running_time = benchmark_compile(model, sample_input, num_iters, backend, None, is_training, optimizer)
+            if running_time is not None:
+                speedup = (eager_time - running_time) / eager_time if eager_time else None
+                table.add_row([("Training" if is_training else "Inference"), backend, "-", compilation_time or "-", running_time, speedup or "-"])
+
+    return table
+
+if __name__ == "__main__":
```

Comment:
This is just convenient for me now locally but will remove everything below this line for merge